frontend/src/pages/dashboard/components/sidebar/components/end-shift-button.tsx

```diff
@@ -0,0 +1,43 @@
+import { Button } from '~/libs/components/components.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useCallback,
+} from '~/libs/hooks/hooks.js';
+import { actions as driverActions } from '~/slices/driver/driver.js';
+import { ShiftStatus } from '~/slices/driver/libs/enums/shift-status.enum';
+import {
+  selectActiveTruck,
+  selectShiftStatus,
+} from '~/slices/driver/selectors.js';
+
+import styles from './styles.module.scss';
+
+const EndShiftButton: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const shiftStatus = useAppSelector(selectShiftStatus);
+  const truck = useAppSelector(selectActiveTruck);
+
+  const isActive = shiftStatus === ShiftStatus.ACTIVE;
+
+  const handleClick = useCallback(() => {
+    void dispatch(driverActions.endShift());
+  }, [dispatch]);
+
+  return isActive ? (
+    <div className={styles.block}>
+      <p className={getValidClassNames(styles.truckInfo, 'textMd')}>
+        {truck?.manufacturer} {truck?.licensePlateNumber}
+      </p>
+      <Button
+        label={'End shift'}
+        frontIcon={'truck'}
+        onClick={handleClick}
+        className={styles.button}
+      />
+    </div>
+  ) : null;
+};
```

Comment:
```suggestion
  return isActive && (
    <div className={styles.block}>
      <p className={getValidClassNames(styles.truckInfo, 'textMd')}>
        {truck?.manufacturer} {truck?.licensePlateNumber}
      </p>
      <Button
        label={'End shift'}
        frontIcon={'truck'}
        onClick={handleClick}
        className={styles.button}
      />
    </div>
  );
};
```
But it seems to me that this button should not check if there is a shift, it is better to do it in the parent component, this logic does not apply to the button, my opinion.
If we want to show the button always, but make it inactive in case of no shift - then we need to use `isDisabled` props for the button. Otherwise we get a component that is rendered but returns nothing if there is no active shift.
And we can pass truck as props, then we don't have to check `shiftStatus === ShiftStatus.ACTIVE` and whether truck has fields.

Reply:
I've replace ternary to &&.
About responsibility of checking the shift. I think Sidebar doesn't have to do that too. It just check the group of the user and render EndSfiftButton component. Then EndSfiftButton check the status of shift and decides to render the button or not.
Let's wait for coach's opinion.

Reply:
If the component should return something, you can use a react fragment. It adds a bit of code, but it still looks cleaner than with a ternary operator. Just wrap the entire return in `<>...</>`.

Reply:
I think we can leave logic for displaying the button here, in this separate component and not to place it inside sidebar
but also rewrite it with react fragment and && as Vlad suggested